code/game/objects/items/props/helmetgarb.dm

```diff
@@ -496,6 +496,32 @@
 	desc = "The USCM had its funding pulled for these when it became apparent that not every deployed enlisted was wearing a helmet 24/7; much to the bafflement of UA High Command."
 	icon_state = "helmet_gasmask"
 
+/obj/item/prop/helmetgarb/helmet_gasmask/on_enter_storage(obj/item/storage/internal/helmet_internal_inventory)
+	if(!istype(helmet_internal_inventory))
+		return
+
+	var/obj/item/clothing/head/helmet/helmet_item = helmet_internal_inventory.master_object
+
+	if(!istype(helmet_item))
+		return
+
+	helmet_item.flags_inventory|=BLOCKGASEFFECT
+	helmet_item.flags_inv_hide|=HIDEFACE
+	..()
+
+/obj/item/prop/helmetgarb/helmet_gasmask/on_exit_storage(obj/item/storage/internal/helmet_internal_inventory)
+	if(!istype(helmet_internal_inventory))
+		return
+
+	var/obj/item/clothing/head/helmet/helmet_item = helmet_internal_inventory.master_object
+
+	if(!istype(helmet_item))
+		return
+
+	helmet_item.flags_inventory = initial(helmet_item.flags_inventory)
+	helmet_item.flags_inv_hide = initial(helmet_item.flags_inv_hide)
+	return..()
```

Comment:
The norm for this proc appears to be to have the parent call first. At the moment with it here your returns earlier will prevent the parent call from occurring.

Move this to be the first line just as `..()` since this proc isn't normally returning anything.